File: haskell-backend/src/Tests/Simple.hs
{-# LANGUAGE DuplicateRecordFields #-}
module Tests.Simple where

import Types
import Data.Set

h1 = User 
  {name="Mark"
  ,countId=1
  ,requirements = fromList [NoSchool, AllowPets]
  ,allowances = fromList [HasSecEight]
  }

h2 = User 
  {name="Sri"
  ,countId=2
  ,requirements = fromList [AllowPets]
  ,allowances = fromList [HasSecEight]
  }

s1 = Space
  {name="New Haven Shelter"
  ,countId=1
  ,requirements = fromList [HasSecEight]
  ,allowances = fromList [NoSchool, AllowPets]
  }

s2 = Space
  {name="Fair Haven Shelter"
  ,countId=2
  ,requirements = fromList [HasSecEight]
  ,allowances = fromList [AllowPets]
  }
  

File: db-testcases/simple.php
<?php
require_once('mongo_config.php');

error_reporting(E_ALL);
ini_set('display_errors', 1);


$c = $db->rooms;
print_r($c->drop());
$c->insert(array("name" => "New Haven Shelter", "requirements" => [ "HasSecEight" ], "allowances" => [ "AllowPets","NoSchool" ], "countId" =>  1 ));
$c->insert(array("name" => "Fair Haven Shelter", "requirements" => [ "HasSecEight" ], "allowances" => [ "AllowPets" ], "countId" =>  2 ));


$c = $db->users;
print_r($c->drop());
$c->insert(array("name" => "Sri", "requirements" => [ "AllowPets" ], "allowances" => [ "HasSecEight" ], "countId" =>  2 ));
$c->insert(array("name" => "Mark", "requirements" => [ "AllowPets","NoSchool" ], "allowances" => [ "HasSecEight" ], "countId" =>  1 ));

//$count = $c->count();
?>
